src/common/utils/http.ts

```diff
@@ -0,0 +1,44 @@
+// ==LICENSE-BEGIN==
+// Copyright 2017 European Digital Reading Lab. All rights reserved.
+// Licensed to the Readium Foundation under one or more contributor license agreements.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file exposed on Github (readium) in the project repository.
+// ==LICENSE-END==
+
+import { container } from "readium-desktop/main/di";
+import { RootState } from "readium-desktop/main/redux/states";
+import { Store } from "redux";
+import * as request from "request";
+import { promisify } from "util";
+
+export type httpResponse = request.Response;
+
+export async function httpGet(url: string, options?: request.CoreOptions): Promise<request.Response> {
+
+    const store = container.get("store") as Store<RootState>;
+    const locale = store.getState().i18n.locale;
+    const requestOptions = Object.assign(
+        {
+            url,
+            method: "GET",
+            encoding: undefined,
+            headers: {
+                "User-Agent": "readium-desktop",
+                "Accept-Language": `${locale},en-US;q=0.7,en;q=0.5`,
+            },
+        },
+        options,
+    );
+    const response = await promisify<request.CoreOptions, request.Response>(request)(requestOptions);
```

Comment:
This does not handle the `error` parameter callback of the original `request` API - which, when wrapped in `util.promisify`, becomes a promise rejection (which must be explicitly captured via `.catch()` or `try {} catch {}`).

Please refer to my proposal again, and see how the edge cases are more explicitly covered. Using `util.promisify` only adds confusion / obfuscation of the original `request` API.

PS: I think `httpGet` can reliably return `Promise<string>`, given `encoding: undefined` (not `null`) in the `request` options. This is currently our main use-case in the codebase, let's not enable `Buffer` for now.

Reply:
- No, it' handle error because promisify throw error with the promise . your example is not explicit either. No ?
- Why promisify add confusion : it's just to return a promise on a function with a callback at the end of the function ?
- Why return body and not response ? It allow to check response inside code and to access directly on body ? Cyrille has begin like this in code.

Reply:
`await` without an explicit `try`/`catch` (or the functional `.catch()` equivalent) is by definition an "unhandled promise rejection" (which there are several in the `readium-desktop` code, unfortunately). Exception handling in the context of promises has different semantics and design patterns than regular cascaded exceptions used to control unexpected side effects.

Reply:
I look at the original `httpGet()` and I see `resolve(body);`, not `resolve(response);` (where `body` is guaranteed to be type `string` is `options.encoding` is `undefined` (not `null`)). I would like the code to continue using this design feature, as it makes it simpler for API consumers. In a future revision of the code, if there is a justification for returning the full HTTP `response`, then we may do it (note that this can be useful for streaming the HTTP payload, in which case the `response.body` is not available in full).

Reply:
- The throw inside an async function return catch(e) with the error in parameter ? No ? I didn't see why you don't like promisify ?

- there was 2 requestGet/httpGEt : https://github.com/readium/readium-desktop/blob/develop/src/main/redux/sagas/update.ts

Reply:
httpGet returns response.body now